src/pages/studyView/charts/ChartContainer.tsx

```diff
@@ -1361,76 +1378,92 @@ export class ChartContainer extends React.Component<IChartContainerProps, {}> {
-                    [styles.highlight]: this.highlightChart,
-                })}
-                data-test={`chart-container-${this.props.chartMeta.uniqueKey}`}
-                onMouseEnter={this.handlers.onMouseEnterChart}
-                onMouseLeave={this.handlers.onMouseLeaveChart}
-            >
-                <ChartHeader
-                    height={this.chartHeaderHeight}
-                    chartMeta={this.props.chartMeta}
-                    chartType={this.props.chartType}
-                    store={this.props.store}
-                    title={this.props.title}
-                    active={this.mouseInChart}
-                    resetChart={this.handlers.resetFilters}
-                    deleteChart={this.handlers.onDeleteChart}
-                    selectedRowsKeys={this.selectedRowsKeys}
-                    toggleLogScale={this.handlers.onToggleLogScale}
-                    toggleLogScaleX={this.handlers.onToggleLogScaleX}
-                    toggleLogScaleY={this.handlers.onToggleLogScaleY}
-                    toggleBoxPlot={this.handlers.onToggleBoxPlot}
-                    toggleViolinPlot={this.handlers.onToggleViolinPlot}
-                    toggleSurvivalPlotLeftTruncation={
-                        this.handlers.onToggleSurvivalPlotLeftTruncation
-                    }
-                    isLeftTruncationAvailable={
-                        this.props.isLeftTruncationAvailable
-                    }
-                    swapAxes={this.handlers.onSwapAxes}
-                    toggleNAValue={this.handlers.onToggleNAValue}
-                    chartControls={this.chartControls}
-                    changeChartType={this.changeChartType}
-                    getSVG={() => Promise.resolve(this.toSVGDOMNode())}
-                    getData={this.props.getData}
-                    downloadTypes={this.props.downloadTypes}
-                    openComparisonPage={this.openComparisonPage}
-                    placement={this.placement}
-                    description={this.props.description}
-                    isCompactSurvivalChart={this.showCompactSurvivalChart}
-                />
-                <div className={styles.chartInnerWrapper}>
-                    {this.props.promise.isPending && (
-                        <LoadingIndicator
-                            isLoading={true}
-                            className={styles.chartLoader}
-                        />
-                    )}
-                    {this.props.promise.isError && (
-                        <div className={styles.chartError}>
-                            <i className="fa fa-warning" aria-hidden="true"></i>{' '}
-                            Error when loading data.
-                        </div>
-                    )}
+            <>
+                {this.alertContent && (
+                    <Modal
+                        show={true}
+                        onHide={() => (this.alertContent = null)}
```

Comment:
not sure if it's a big deal but this function can also be wrapped with `action`, right?